src/main/java/bio/terra/cli/serialization/disk/DiskConfig.java

```diff
@@ -0,0 +1,93 @@
+package bio.terra.cli.serialization.disk;
+
+import static bio.terra.cli.Config.BrowserLaunchOption;
+import static bio.terra.cli.Config.CommandRunnerOption;
+
+import bio.terra.cli.Config;
+import bio.terra.cli.utils.Logger;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+
+/**
+ * External representation of a configuration for writing to disk.
+ *
+ * <p>This is a POJO class intended for serialization. This JSON format is not user-facing.
+ *
+ * <p>See the {@link Config} class for a configuration's internal representation.
+ */
+@JsonDeserialize(builder = DiskConfig.Builder.class)
+public class DiskConfig {
+  public final BrowserLaunchOption browserLaunchOption;
+  public final CommandRunnerOption commandRunnerOption;
+  public final String dockerImageId;
+  public final int resourcesCacheSize;
+  public final Logger.LogLevel fileLoggingLevel;
+  public final Logger.LogLevel consoleLoggingLevel;
+
+  /** Serialize an instance of the internal class to the disk format. */
+  public DiskConfig(Config internalObj) {
+    this.browserLaunchOption = internalObj.getBrowserLaunchOption();
```

Comment:
Consider avoiding circular dependencies between classes. As-is, DiskConfig depends on Config and Config depends on DiskConfig, with mutual constructors. Instead, have a acyclic dependency, where only DiskConfig depends on Config by having its contructor take the Config and also having a `toConfig` method to call a generic Config constructor.
Serialization classes should be aware of the business logic class, but generally not vice versa.

ditto other "Disk" and "Command" objects

Reply:
The thing that makes me hesitate to move the conversions from the internal classes to the serialization classes is that I think we wouldn't be able to take advantage of the `Resource` classes inheritance.
e.g. If we're looping through a list of `Resource`s to write them to disk,  we can either call a method on the `Resource` object which knows what `DiskResource` constructor to call (`BqDataset.serializeToDisk` knows to return a `DiskBqDataset`), or we can call a generic method on the `DiskResource` class, which would need a case statement to decide which `DiskResource` sub-class constructor to call for each resource. This gives me pause because the `Resource` classes are setup with inheritance specifically to avoid these case statements that used to be all over the place.

Reply:
Good point. I think I'd still be inclined to break the circular dependency then by making the serialization classes not aware of the business objects and having the business objects populate all the fields on the serialization object builders, but optional.

Reply:
OK, going to leave as is for now. We might want to re-think this if more serialization formats come into the picture.